vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/VoiceBroadcastItemFactory.kt

```diff
@@ -15,46 +15,93 @@
 import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.util.toMatrixItem
 import javax.inject.Inject
 
 class VoiceBroadcastItemFactory @Inject constructor(
         private val session: Session,
         private val avatarSizeProvider: AvatarSizeProvider,
-        private val audioMessagePlaybackTracker: AudioMessagePlaybackTracker,
+        private val colorProvider: ColorProvider,
+        private val drawableProvider: DrawableProvider,
+        private val voiceBroadcastRecorder: VoiceBroadcastRecorder?,
+        private val voiceBroadcastPlayer: VoiceBroadcastPlayer,
 ) {
 
     fun create(
+            params: TimelineItemFactoryParams,
             messageContent: MessageVoiceBroadcastInfoContent,
-            eventsGroup: TimelineEventsGroup?,
             highlight: Boolean,
             callback: TimelineEventController.Callback?,
             attributes: AbsMessageItem.Attributes,
-    ): MessageVoiceBroadcastItem? {
+    ): VectorEpoxyModel<out VectorEpoxyHolder>? {
         // Only display item of the initial event with updated data
         if (messageContent.voiceBroadcastState != VoiceBroadcastState.STARTED) return null
-        val voiceBroadcastEventsGroup = eventsGroup?.let { VoiceBroadcastEventsGroup(it) } ?: return null
+        val eventsGroup = params.eventsGroup ?: return null
+        val voiceBroadcastEventsGroup = VoiceBroadcastEventsGroup(eventsGroup)
         val mostRecentTimelineEvent = voiceBroadcastEventsGroup.getLastDisplayableEvent()
         val mostRecentEvent = mostRecentTimelineEvent.root.asVoiceBroadcastEvent()
         val mostRecentMessageContent = mostRecentEvent?.content ?: return null
         val isRecording = mostRecentMessageContent.voiceBroadcastState != VoiceBroadcastState.STOPPED && mostRecentEvent.root.stateKey == session.myUserId
-        return MessageVoiceBroadcastItem_()
+        return if (isRecording) {
+            createRecordingItem(params.event.roomId, highlight, callback, attributes)
+        } else {
+            createListeningItem(params.event.roomId, eventsGroup.groupId, highlight, callback, attributes)
+        }
+    }
+
+    private fun createRecordingItem(
+            roomId: String,
+            highlight: Boolean,
+            callback: TimelineEventController.Callback?,
+            attributes: AbsMessageItem.Attributes,
+    ): MessageVoiceBroadcastRecordingItem {
+        val roomSummary = session.getRoom(roomId)?.roomSummary()
+        return MessageVoiceBroadcastRecordingItem_()
+                .attributes(attributes)
+                .highlighted(highlight)
+                .roomItem(roomSummary?.toMatrixItem())
+                .colorProvider(colorProvider)
+                .drawableProvider(drawableProvider)
+                .voiceBroadcastRecorder(voiceBroadcastRecorder)
+                .leftGuideline(avatarSizeProvider.leftGuideline)
+                .callback(callback)
```

Comment:
You need to call `.id(...)`. There are lots of `...Item_()` without id in the project though, this is strange.